rust-runtime/aws-smithy-http-auth/src/auth.rs

```diff
@@ -187,11 +223,61 @@ impl HttpAuthDefinition {
     }
 }
 
+/// Types associated with [`HttpAuthDefinition`].
+pub mod http_auth_definition {
+    use super::AuthError;
+    use super::AuthErrorKind;
+    use super::HttpAuthDefinition;
+    use super::HttpAuthLocation;
+
+    /// A builder for [`HttpAuthDefinition`].
+    #[derive(Debug, Default)]
+    pub struct Builder {
+        location: HttpAuthLocation,
+        name: String,
+        scheme: Option<String>,
+    }
+
+    impl Builder {
+        /// Sets the HTTP auth location.
+        pub fn location(mut self, location: HttpAuthLocation) -> Self {
+            self.location = location;
+            self
+        }
+
+        /// Sets the the HTTP auth name.
+        pub fn name(mut self, name: impl Into<String>) -> Self {
+            self.name = name.into();
+            self
+        }
+
+        /// Sets the HTTP auth scheme.
+        pub fn scheme(mut self, scheme: impl Into<Option<String>>) -> Self {
+            let scheme: Option<String> = scheme.into();
+            self.scheme = scheme;
+            self
+        }
+
+        /// Constructs a [`HttpAuthDefinition`] from the builder.
+        pub fn build(self) -> Result<HttpAuthDefinition, AuthError> {
+            if self.scheme.is_some() && !self.name.eq_ignore_ascii_case("authorization") {
+                return Err(AuthError::from(AuthErrorKind::SchemeNotAllowed));
```

Comment:
Seems to me like this should panic instead. This seems like encountering this would be programmer error rather than caused by user input.

Reply:
That would be fine.